cdap-ui/app/cdap/components/AbstractWidget/KeyValueDropdownWidget/KeyValueDropdownRow.tsx

```diff
@@ -108,37 +116,59 @@ class KeyValueDropdownRow extends AbstractRow<IKeyValueDropdownRowProps, IKeyVal
       };
     });
 
-    return (
-      <div className={this.props.classes.inputContainer}>
+    const InputField = (props) => {
+      return (
         <Input
           classes={{ disabled: this.props.classes.disabled }}
           placeholder={this.props.keyPlaceholder}
-          onChange={this.handleChange.bind(this, 'key')}
-          value={this.state.key}
+          onChange={this.handleChange.bind(this, props.type)}
+          value={props.value}
           autoFocus={this.props.autofocus}
           onKeyPress={this.handleKeyPress}
           onKeyDown={this.handleKeyDown}
           disabled={this.props.disabled}
           inputRef={this.props.forwardedRef}
-          data-cy="key"
+          data-cy={props.type}
         />
+      );
+    };
 
+    const SelectField = (props) => {
```

Comment:
This won't work. When we define this as another component and pass in props we React will re-render this on each change as the props are different. That will make the focus lost from the current textbox. 

Was this change tested before being pushed?

Reply:
Please replace this change with just variables. We should not define this as another stateless react components. Also We shouldn't be mixing components (`props` and `this.props`)?

Reply:
Replace the snippet with
```
const inputType = this.props.ordering === OrderingEnum.KEYSFIRST ? 'key' : 'value';
    const InputField = (
      <Input
        classes={{ disabled: this.props.classes.disabled }}
        placeholder={this.props.keyPlaceholder}
        onChange={this.handleChange.bind(this, inputType)}
        value={this.props.ordering === OrderingEnum.KEYSFIRST ? this.state.key : this.state.value}
        autoFocus={this.props.autofocus}
        onKeyPress={this.handleKeyPress}
        onKeyDown={this.handleKeyDown}
        disabled={this.props.disabled}
        inputRef={this.props.forwardedRef}
        data-cy={inputType}
      />
    );

    const selectType = this.props.ordering === OrderingEnum.VALUESFIRST ? 'key' : 'value';
    const SelectField = (
      <Select
        classes={{ disabled: this.props.classes.disabled }}
        value={this.props.ordering === OrderingEnum.VALUESFIRST ? this.state.key : this.state.value}
        onChange={this.handleChange.bind(this, selectType)}
        displayEmpty={true}
        disabled={this.props.disabled}
        data-cy={selectType}
      >
        {dropdownOptions.map((option) => (
          <MenuItem
            value={option.value}
            key={option.value}
            data-cy={`${selectType}-${option.value}`}
          >
            {option.label}
          </MenuItem>
        ))}
      </Select>
    );```